src/features/defi/providers/foxy/components/FoxyManager/Deposit/components/Confirm.tsx

```diff
@@ -98,16 +98,18 @@ export const Confirm: React.FC<ConfirmProps> = ({ onNext, accountId }) => {
 
       const transactionReceipt = await poll({
         fn: () => foxyApi.getTxReceipt({ txid }),
-        validate: (result: ethers.providers.TransactionReceipt) => !isNil(result),
+        validate: (result: TransactionReceipt | null) => !isNil(result),
         interval: 15000,
         maxAttempts: 30,
       })
       dispatch({
         type: FoxyDepositActionType.SET_DEPOSIT,
         payload: {
-          txStatus: transactionReceipt.status ? 'success' : 'failed',
-          usedGasFeeCryptoBaseUnit: transactionReceipt.effectiveGasPrice
-            .mul(transactionReceipt.gasUsed)
+          txStatus: transactionReceipt?.status ? 'success' : 'failed',
+          usedGasFeeCryptoBaseUnit: bnOrZero(
+            (transactionReceipt as TransactionReceiptParams | null)?.effectiveGasPrice?.toString(),
+          )
+            .times(transactionReceipt?.gasUsed.toString() ?? 0)
```

Comment:
Is coelescing necessary here? Because other cases dont

https://github.com/shapeshift/web/pull/6224/files#diff-4cea5d5534d77c89e34e8aa387250f9caa12d81a387293f207f90dff42676bebR118

Reply:
We use `bnOrZero` in the other version, same thing different syntax!